x-pack/filebeat/input/streaming/websocket.go

```diff
@@ -116,6 +116,7 @@ func (s *websocketStream) FollowStream(ctx context.Context) error {
 			return ctx.Err()
 		default:
 			_, message, err := c.ReadMessage()
+			s.metrics.receivedBytesTotal.Add(uint64(len(message)))
```

Comment:
Why move this up? Is a message valid when `err` is non-nil?

Reply:
Shouldn't we still count the received bytes irrespective of If it's a valid message or not, since this is metric ?

Reply:
I think this depends on what the understood semantics of the metric are.

Reply:
As this is not a core part of the change for this PR, I've reverted this now, will take a look at it once more when I do some cleanups.